lib/ansible/modules/cloud/amazon/kinesis_stream.py

```diff
@@ -695,14 +700,61 @@ def retention_action(client, stream_name, retention_period=24,
     return success, err_msg
 
 
-def update(client, current_stream, stream_name, retention_period=None,
+def update_shard_count(client, stream_name, number_of_shards=1, check_mode=False):
+    """Increase or Decrease the number of shards in the Kinesis stream.
+    Args:
+        client (botocore.client.EC2): Boto3 client.
+        stream_name (str): The name of the kinesis stream.
+
+    Kwargs:
+        number_of_shards (int): Number of shards this stream will use.
+            default=1
+        check_mode (bool): This will pass DryRun as one of the parameters to the aws api.
+            default=False
+
+    Basic Usage:
+        >>> client = boto3.client('kinesis')
+        >>> stream_name = 'test-stream'
+        >>> number_of_shards = 3
+        >>> update_shard_count(client, stream_name, number_of_shards)
+
+    Returns:
+        Tuple (bool, str)
+    """
+    success = False
+    err_msg = ''
+    params = {
+        'StreamName': stream_name,
+        'ScalingType': 'UNIFORM_SCALING'
+    }
+    try:
+        if not check_mode:
+            params['TargetShardCount'] = number_of_shards
+            res = client.update_shard_count(**params)
+            success = True
+            err_msg = (
+                'Number of shards changed successfully from {0} to {1}'
+                .format(res['CurrentShardCount'], res['TargetShardCount'])
+            )
+        else:
+            success = True
+
+    except botocore.exceptions.ClientError as e:
+        err_msg = str(e)
```

Comment:
@mikedlr I agree that these need to change. Right now I don't think we're able to add traceback. This is just returning the error. Do you think it makes me sense to fix in a separate PR? Because *all* of them need to change. All of the functions are just taking the error message and passing it along. But a problem is that the messages are being overwritten in a number of places...